tools/stress-cluster/cluster/kubernetes/stress-infrastructure/templates/stresswatcher.yaml

```diff
@@ -27,12 +30,16 @@ spec:
       - name: stresswatcher
         imagePullPolicy: Always
         image: azsdkengsys.azurecr.io/stress/watcher:{{ .Values.tag }}
-        command: ["dotnet", "Stress.Watcher.dll"]
+        command: ["sh", "-c"]
+        args:
+          - >
+            az login --federated-token "$(cat $AZURE_FEDERATED_TOKEN_FILE)" --service-principal -u $AZURE_CLIENT_ID -t $AZURE_TENANT_ID;
```

Comment:
Is using a file the only real way to share this token?

Reply:
@weshaggard The file is the implementation method of AKS for federated auth, so unfortunately yes.